File: 965-univalued-binary-tree/965-univalued-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isUnivalTree = function(root) {
    let value = root.val;
    return helper(root, value);
};

var helper = function(root, value){
        if(root == null) return true;
        if(root.val != value) return false;
        return helper(root.left, value) && helper(root.right, value);
    }

File: 965-univalued-binary-tree/965-univalued-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isUnivalTree(TreeNode root) {
        int value = root.val;
        return helper(root, value);
    }
    
    public boolean helper(TreeNode root, int value){
        if(root == null) return true;
        if(root.val != value) return false;
        return helper(root.left, value) && helper(root.right, value);
    }
}